lib/checkers/zookeeper_health_checker.rb

```diff
@@ -66,6 +73,21 @@ def check
 
   private
 
+  def active_health_check(check_details)
+    node_path = zk_client.create("/#{hostname}", ephemeral: true, sequential: true)
+    zk_client.delete(node_path)
+
+    true
+  rescue => e
+    logger.error { "Caught error during health check: #{e.inspect}\n#{e.backtrace.join("\n")}" }
+    false
+  end
+
+  def zk_client
+    @zk_client ||= ZK.new("#{zk_connection_settings[:host]}:#{zk_connection_settings[:port]}/whazzup",
```

Comment:
Yep, the ```zk``` gem seems to handle the reconnect logic well. I think there might be a small window where whazzup can leak ephemeral nodes because the ZK client will resume the same session if it hasn't timed out, but I don't think it's worth worrying about since they will go away after rebooting either whazzup or ZK, and trying to keep them from leaking will require extra logic for little gain.